include/SDL3/SDL_properties.h

```diff
@@ -0,0 +1,133 @@
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+
+/**
+ *  \file SDL_properties.h
+ *
+ *  \brief Header file for SDL properties.
+ */
+
+#ifndef SDL_properties_h_
+#define SDL_properties_h_
+
+#include <SDL3/SDL_begin_code.h>
+/* Set up for C function definitions, even when using C++ */
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * SDL properties ID
+ */
+typedef Uint32 SDL_PropertiesID;
+
+/**
+ * Create a set of properties
+ *
+ * \returns an ID for a new set of properties, or 0 on failure; call SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_DestroyProperties
+ */
+extern DECLSPEC SDL_PropertiesID SDLCALL SDL_CreateProperties(void);
+
+/**
+ * Lock a set of properties
+ *
+ * Obtain a multi-threaded lock for these properties. Other threads will wait while trying to lock these properties until they are unlocked. Properties must be unlocked before they are destroyed.
+ *
+ * The lock is automatically taken when setting individual properties, this function is only needed when you want to set several properties atomically or want to guarantee that properties being queried aren't freed in another thread.
+ *
+ * \param props the properties to lock
+ *
+ * \returns 0 on success or a negative error code on failure; call
+ *          SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_UnlockProperties
+ */
+extern DECLSPEC int SDLCALL SDL_LockProperties(SDL_PropertiesID props);
+
+/**
+ * Unlock a set of properties
+ *
+ * \param props the properties to unlock
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_LockProperties
+ */
+extern DECLSPEC void SDLCALL SDL_UnlockProperties(SDL_PropertiesID props);
+
+/**
+ * Set a property on a set of properties
+ *
+ * \param props the properties to modify
+ * \param name the name of the property to modify
+ * \param value the new value of the property, or NULL to delete the property
+ * \param cleanup the function to call when this property is deleted, or NULL if no cleanup is necessary
+ * \param userdata a pointer that is passed to the cleanup function
+ *
+ * \returns 0 on success or a negative error code on failure; call
+ *          SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_GetProperty
+ */
+extern DECLSPEC int SDLCALL SDL_SetProperty(SDL_PropertiesID props, const char *name, void *value, void (*cleanup)(void *userdata, void *value), void *userdata);
```

Comment:
Should the `cleanup` procedure be `SDLCALL`, e.g. `void (SDLCALL *cleanup)(....)`, like the others in SDL? (Not that it matters for now, but for correctness?)

Reply:
Ah, yes, it should.